wgengine/userspace.go

```diff
@@ -512,38 +512,9 @@ func (e *userspaceEngine) pollResolver() {
 			continue
 		}
 
-		var buf []byte
 		const offset = tstun.PacketStartOffset
-		switch {
-		case to.IP().Is4():
-			h := packet.UDP4Header{
-				IP4Header: packet.IP4Header{
-					Src: magicDNSIP,
-					Dst: to.IP(),
-				},
-				SrcPort: magicDNSPort,
-				DstPort: to.Port(),
-			}
-			hlen := h.Len()
-			// TODO(dmytro): avoid this allocation without importing tstun quirks into dns.
-			buf = make([]byte, offset+hlen+len(bs))
-			copy(buf[offset+hlen:], bs)
-			h.Marshal(buf[offset:])
-		case to.IP().Is6():
-			h := packet.UDP6Header{
-				IP6Header: packet.IP6Header{
-					Src: magicDNSIPv6,
-					Dst: to.IP(),
-				},
-				SrcPort: magicDNSPort,
-				DstPort: to.Port(),
-			}
-			hlen := h.Len()
-			// TODO(dmytro): avoid this allocation without importing tstun quirks into dns.
```

Comment:
I guess we're just accepting this allocation at this point? :)

Reply:
(I mean, it's DNS traffic and not super high volume so that's probably fine)

Reply:
Given its a straightforward change, I've updated the code to read the offset and allocate enough space originally. I had a quick look through `tstun`, it always gets imported anyway so I don't see any downside to reading the const straight from the package.